File: CS-Examples/06_Comments/SetCommentFillColor.cs
using Spire.Xls;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace SetCommentFillColor
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void btnRun_Click(object sender, EventArgs e)
        {
            // Create a workbook
            Workbook workbook = new Workbook();

            // Get the first worksheet
            Worksheet sheet = workbook.Worksheets[0];

            // Create Excel font
            ExcelFont font = workbook.CreateFont();
            font.FontName = "Arial";
            font.Size = 11;
            font.KnownColor = ExcelColors.Orange;

            // Add the comment
            CellRange range = sheet.Range["A1"];
            range.Comment.Text = "This is a comment";
            range.Comment.RichText.SetFont(0, (range.Comment.Text.Length - 1), font);

            // Set comment Color
            range.Comment.Fill.FillType = ShapeFillType.SolidColor;
            range.Comment.Fill.ForeColor = Color.SkyBlue;

            // Set comment is visible
            range.Comment.Visible = true;

            // Specify the resulting file name
            String result = "SetCommentFillColor_out.xlsx";

            // Save the modified workbook to a file using Excel 2013 format
            workbook.SaveToFile(result, ExcelVersion.Version2013);

            // Dispose of the workbook object to release resources
            workbook.Dispose();

            //View the document
            FileViewer(result);
        }

        private void FileViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }
    }
}

File: VB-Examples/06_Comments/SetCommentFillColor.vb
Imports Spire.Xls
Imports System.ComponentModel
Imports System.Text

Namespace SetCommentFillColor
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            'Creates a new instance of the Workbook class.
            Dim workbook As New Workbook()

            'Gets the first worksheet from the workbook.
            Dim sheet As Worksheet = workbook.Worksheets(0)

            'Create a new Excel font.
            Dim font As ExcelFont = workbook.CreateFont()
            font.FontName = "Arial"
            font.Size = 11
            font.KnownColor = ExcelColors.Orange

            'Get the CellRange object for cell A1.
            Dim range As CellRange = sheet.Range("A1")

            'Set the comment text for the cell.
            range.Comment.Text = "This is a comment"

            'Set the font for the comment's rich text.
            range.Comment.RichText.SetFont(0, (range.Comment.Text.Length - 1), font)

            'Set the fill type of the comment's shape fill to solid color.
            range.Comment.Fill.FillType = ShapeFillType.SolidColor

            'Set the foreground color (fill color) for the comment's shape.
            range.Comment.Fill.ForeColor = Color.SkyBlue

            'Set the visibility of the comment to true.
            range.Comment.Visible = True

            'Specify the filename to save the modified workbook.
            Dim result As String = "SetCommentFillColor_out.xlsx"

            'Save the workbook to the specified file using Excel 2013 format.
            workbook.SaveToFile(result, ExcelVersion.Version2013)
            ' Release the resources used by the workbook
            workbook.Dispose()

            'View the document
            FileViewer(result)
		End Sub

		Private Sub FileViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub
	End Class
End Namespace
